core/src/main/scala/akka/kafka/ConsumerActorTerminatedException.scala

```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.kafka
+
+class ConsumerActorTerminatedException extends Exception("Consumer actor terminated")
```

Comment:
I agree a raw exception is a bit crude, on the other hand this seems rather specific and 'leaks implementation details' - allowing you to match on this specific exception makes it harder for us to refactor safely later.

Could we come up with a broader category of problems that we could put this exception under? What was the context in which you wanted to treat the "Consumer actor terminated" problem differently from any other problems?

Reply:
I agree with both. We should have a more specific exception, but still name it less implementation specific.
`ConsumerFailed`?

It should be a `RuntimeException`.

Reply:
A more general and less implementation specific name makes sense to me. Unfortunately I think we have to leave the message the same because there's probably already plenty of code out there with `if ex.getMessage.contains("Consumer actor terminated") => ...`. I like `ConsumerFailed`.